### ISSUE TITLE
1.4.10 Reflow:  Admin Dashboard's responsiveness

### ISSUE THREAD

**Issue by orangewolf** [scientist-softserv/utk-hyku#41]
ref:
- #407 
- #393 

In addition to this ticket, please create an issue for this in [Hyku](https://github.com/samvera/hyku/issues) and link it to this ticket.

# Summary

The admin dashboard failed the accessibility audit because, at the tested dimensions described below, a website should adjust itself to be viewable vertically. A user should be able to see and use a vertical scrollbar, but not a horizontal one. 

We've determined that this currently does not work because the admin dashboard has a left sidebar that does not collapse by default. Once we manually collapse the sidebar, it passes the audit test. 

Some solution options may include: 

- [ ] When visiting the admin dashboard page, the side bar should be collapsed by default. All users must manually expand the sidebar to view its options. 

  Or, with CSS the above would only happen after certain dimensions are met. (please see the notes section)

For inspiration + things to consider: 

Gitlab passes this requirement! Notice how the left side bar is collapsed, but when you hover over the icons the additional options are revealed. 

![Image](https://user-images.githubusercontent.com/29311858/203172232-d9c3ea42-7b94-4561-80c3-b7ec94913485.png)

Currently, when you hover over the collapsed icons in hyku, the user really has no idea what their significance is. See demo: https://share.getcloudapp.com/xQuw9yKb

# Acceptance Criteria

- ~~[ ] Viewing the admin dashboard at 320 CSS pixels (see notes section) should pass the accessibility audit. The website should be responsive and align it self for vertical viewing.~~
- ~~[ ] no horizontal scrolling should be available~~
- ~~[ ] Please reference this document for additional details, section 1.4.10 Reflow: [Report for https___test.hykucommons.org_.pdf](https://github.com/scientist-softserv/utk-hyku/files/10061210/Report.for.https___test.hykucommons.org_.pdf)~~
- [x] Collapse sidebar at a specific width
- [x] Add title on hover over to the sidebar icons

# Testing Instructions (On desktop only)
1. Visit the UTK staging site. 
1. Login as an admin user. 
1. Go to the dashboard. 
1. Adjust your viewing settings to meet the dimensions described below. 
1. Check the Acceptance Criteria for pass/fail

### How to adjust settings: 
1. right-click inspect to adjust width: https://share.getcloudapp.com/Wnu7ARdA
1. Zoom to 400% by holding command btn and + several times. Hold cmd and - to zoom out.

# Notes

This ticket has been created as a result of an accessibility audit: ref: [Report for https___test.hykucommons.org_.pdf](https://github.com/scientist-softserv/utk-hyku/files/10061211/Report.for.https___test.hykucommons.org_.pdf)

**NOTE: The following pages were tested at 320 CSS pixels, which is equivalent to a starting viewport width of 1280 CSS pixels wide at 400% zoom.**

**Comment by DiemBTran** [comment]
Linked items:

- EPIC: Accessibility Improvements

**Comment by DiemBTran** [comment]
Kirk June 2022:

---

Potentially reach out to Samvera community to see if they have a solution for this.

**Comment by DiemBTran** [comment]
LaRita June 2022:

---

Notes regarding design: 

If we set the selection to minimize automatically, then we probably want some sort of hover menu rather than the current dropdown menus for standard sized views. Additionally, we need to make sure that if it starts as minimized, once it is toggled, it remains maximized until the user minimizes it again. Currently just removing the maximized keyword isn’t adequate.

Using hover menus means we need a different menu structure for touch screens where hover isn’t an option. The ticket shows gitlab menus as an example. If you view those on a phone, they switch to a hamburger menu structure rather than the hover they use for standard views.

**Comment by DiemBTran** [comment]
Shana - June 2022

---

Discussed during team planning mtg: 

- [ ] At a certain size cut off/media query, sidebar menu should be collapsed and stay collapsed. 
  - [ ] investigate the query cut off responsible for turning the top bar into a hamburger icon. (760px wide and below). Use the same.
  - [ ] how does the top bar go back in forth between an expanded and hamburger?
  - [ ] ![Image](https://user-images.githubusercontent.com/29311858/203176122-90b8ead9-1d9e-45d3-8cef-fc6005e9e534.png)
  - [ ] ![Image](https://user-images.githubusercontent.com/29311858/203176132-ca7e4398-3370-475c-b8f6-60c45e42da3a.png)


- [ ] Once collapsed, only icons will show. On hov, display the text of that menu icon.

**Comment by jillpe** [comment]
related to #385

**Comment by jillpe** [comment]
Separate ticket created for first 3 acceptance criteria
- #485

**Comment by jillpe** [comment]
SoftServ QA: Pass!

 <details><summary>Collapse sidebar at a specific width</summary>
Collapses at 150%


https://github.com/scientist-softserv/utk-hyku/assets/84697174/a78c30da-578b-47c0-a336-771710c29c68


</details>


 <details><summary>Add title on hover over to the sidebar icons</summary>


https://github.com/scientist-softserv/utk-hyku/assets/84697174/5f940203-ffea-42a0-85ea-2050a3cccdbd


</details>

**Comment by josh-morgan117** [comment]
This isn't working for me in staging on Chrome or Firefox. The arrow up top to collapse the sidebar doesn't even work.

**Comment by ShanaLMoore** [comment]
We have a fix coming shortly. Please put this on hold and anything regarding sidebar/header drop downs. A recent update broke this behavior. 

ref: https://github.com/scientist-softserv/utk-hyku/issues/491

**Comment by ShanaLMoore** [comment]
unblocked, the bug as been resolved. cc @kidon0011

**Comment by josh-morgan117** [comment]
Thanks, works for me now.
